### DISCUSSION TITLE
Add "Multiple Users Editing Page" Warning

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#4793] [Feature Requests]
If you navigate to an Item Detail page, and there is already a Directus user on that page, you should get a notification to _be careful_. Let's use the same warning text as v8 for now.

**Comment by rijkvanzanten** [comment]
Shall we make this a dialog instead of a notification? The notification might be easy to miss, while it's a pretty big "HEY PROCEED WITH CAUTION" kinda thing

**Comment by benhaynes** [comment]
Hmm, we could, but I worry that in a platform with a lot of "visualized" (readonly) data, that this could be annoying though. Imagine analytics where it's all readonly and you have many users going around looking at things. A blocking dialog might be a bit much... would we want this configurable at the role or collection level?

**Comment by w0kyj** [comment]
At a minimum need to warn a user when they commit that someone else has committed a change already.
- Allow user to cancel or overwrite(note revisions will be tracked)
    - even better allow a merge operation - show other users changes compared to your changes ...

Regarding pre-warning (only need to notify if another user with edit permissions accesses the same record).

Other thoughts/considerations:
- Make edit first come, locks out
    - make read-only for other users until current users commits and close
    - Only make read-only after the first user makes an edit
        - Make read-only for everyone until commit, then force refresh all editors(minimum) and unlock
    - lots of admin headaches with this and would need a way to boot a user if they accidentally keep a record open or in edit
- If a user commits a change, then notify and lock any other edit users save until they refresh
    - No idea on complexities here

**Comment by bcartier** [comment]
Not sure about a locking system - that could really cause headaches. 

I do really like the idea of a warning however. Maybe the warning should tie into the permissions system so that it's only triggered if someone is on a detail page for an item that they have permission to edit?

**Reply by benhaynes** [reply]
I agree. The first thing I thought of was someone leaving work for the weekend ... with their account on a page / with field focus. We wouldn't want to lock people out accidentally!

**Reply by jordie23** [reply]
Wordpress solves this by having a "Take Over" button, so no one is truely locked out, but they are made very aware something might be lost if they proceed.

**Comment by jordie23** [comment]
Is there any feature like this planned? I had a play and couldn't see even a message that someone else is on the page.

**Reply by azrikahar** [reply]
This is not an existing feature yet, but it can definitely be looked into as an enhancement to the App after Websockets gets added to the API via #14737.

**Comment by lluishi93** [comment]
hi, this was an existing feature in Directus 8. For large projects with multiple people editing collections at the same time, this is much needed in order to have proper coordination. Is there any plans to add something like this?

**Reply by rijkvanzanten** [reply]
The plan is actually to go one step further. https://github.com/directus/directus/pull/14737 is an important first step

**Reply by lluishi93** [reply]
> The plan is actually to go one step further. #14737 is an important first step

looking forward it!

**Reply by LevelUpGit** [reply]
Hi, is there any progress on this?

**Comment by licitdev** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
